feature-settings-impl/src/main/java/io/novafoundation/nova/feature_settings_impl/presentation/networkManagement/networkList/common/adapter/NetworkManagementListAdapter.kt

```diff
@@ -0,0 +1,48 @@
+package io.novafoundation.nova.feature_settings_impl.presentation.networkManagement.networkList.common.adapter
+
+import android.annotation.SuppressLint
+import android.view.ViewGroup
+import androidx.recyclerview.widget.DiffUtil
+import androidx.recyclerview.widget.ListAdapter
+import coil.ImageLoader
+import io.novafoundation.nova.feature_settings_impl.presentation.networkManagement.networkList.common.adapter.items.NetworkListNetworkRvItem
+import io.novafoundation.nova.feature_settings_impl.presentation.networkManagement.networkList.common.adapter.items.NetworkListNetworkViewHolder
+import io.novafoundation.nova.feature_settings_impl.presentation.networkManagement.networkList.common.adapter.items.NetworkListRvItem
+import io.novafoundation.nova.feature_settings_impl.presentation.networkManagement.networkList.common.adapter.items.NetworkListViewHolder
+
+private const val NETWORK_VIEW_TYPE = 0
+
+class NetworkManagementListAdapter(
+    private val imageLoader: ImageLoader
+) : ListAdapter<NetworkListRvItem, NetworkListViewHolder>(NetworkManagementListDiffCallback()) {
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): NetworkListViewHolder {
+        return when (viewType) {
+            NETWORK_VIEW_TYPE -> NetworkListNetworkViewHolder(parent, imageLoader)
+            else -> throw IllegalStateException("Unknown view type: $viewType")
```

Comment:
Why do we need multiple view types switch with only one view type present?

Reply:
It's for future. We will use item "Add network" later